dbt/include/sqlserver/macros/adapters.sql

```diff
@@ -72,31 +72,16 @@
 {% endmacro %}
 
 {% macro sqlserver__drop_relation(relation) -%}
-  {% if relation.type == 'view' -%}
-   {% set object_id_type = 'V' %}
-   {% elif relation.type == 'table'%}
-   {% set object_id_type = 'U' %}
-   {%- else -%} invalid target name
-   {% endif %}
-  {% call statement('drop_relation', auto_begin=False) -%}
-    if object_id ('{{ relation.include(database=False) }}','{{ object_id_type }}') is not null
-      begin
-      drop {{ relation.type }} {{ relation.include(database=False) }}
-      end
+  {% call statement('drop_relation', auto_begin=False) %}
+    USE [{{ relation.database }}]
+    DROP VIEW IF EXISTS {{ relation.include(database=False) }}
+    DROP TABLE IF EXISTS {{ relation.include(database=False) }}
```

Comment:
the reason we're using the `object_id()` calls is because `IF EXISTS` wasn't added to SQL Server until 2016, which @mikaelene already mentioned. Can you see a way of getting the `USE` statement to play nicely with the `IF object_id()` piece?